edge-modules/TwinTester/TwinOperator.cs

```diff
@@ -0,0 +1,106 @@
+        }
+
+        static int GetNewPropertyCounter(Dictionary<string, DateTime> properties)
+        {
+            int maxPropertyId = -1;
+            foreach (KeyValuePair<string, DateTime> propertyUpdate in properties)
+            {
+                maxPropertyId = Math.Max(int.Parse(propertyUpdate.Key), maxPropertyId);
+            }
+
+            return maxPropertyId + 1;
+        }
+
+        static TwinCollection GetReportedPropertiesResetTwin(Twin originalTwin)
+        {
+            TwinCollection eraseReportedProperties = new TwinCollection();
+            foreach (dynamic twinUpdate in originalTwin.Properties.Reported)
+            {
+                KeyValuePair<string, object> pair = (KeyValuePair<string, object>)twinUpdate;
+                eraseReportedProperties[pair.Key] = null; // erase twin property by assigning null
+            }
+
+            return eraseReportedProperties;
+        }
+
+        public static async Task<TwinState> InitializeModuleTwinAsync(RegistryManager registryManager, ModuleClient moduleClient, TwinEventStorage storage)
+        {
+            try
+            {
+                TwinState initializedState;
+                Twin twin = await registryManager.GetTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId);
+                int storageCount = (await storage.GetAllDesiredPropertiesReceivedAsync()).Count + (await storage.GetAllDesiredPropertiesUpdatedAsync()).Count + (await storage.GetAllReportedPropertiesUpdatedAsync()).Count;
+                if (storageCount == 0)
+                {
+                    Logger.LogInformation("No existing storage detected. Initializing new module twin for fresh run.");
+
+                    // reset desired properties
+                    Twin desiredPropertyResetTwin = await registryManager.ReplaceTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId, new Twin(), twin.ETag);
+
+                    // reset reported properties
+                    TwinCollection eraseReportedProperties = GetReportedPropertiesResetTwin(desiredPropertyResetTwin);
+                    await moduleClient.UpdateReportedPropertiesAsync(eraseReportedProperties);
+
+                    await Task.Delay(TimeSpan.FromSeconds(10)); // give ample time for reported properties reset to reach cloud
+                    twin = await registryManager.GetTwinAsync(Settings.Current.DeviceId, Settings.Current.ModuleId);
+                    initializedState = new TwinState { ReportedPropertyUpdateCounter = 0, DesiredPropertyUpdateCounter = 0, TwinETag = twin.ETag, LastTimeOffline = DateTime.MinValue };
+                }
+                else
+                {
+                    Logger.LogInformation("Existing storage detected. Initializing reported / desired property update counters.");
+                    Dictionary<string, DateTime> reportedProperties = await storage.GetAllReportedPropertiesUpdatedAsync();
```

Comment:
It should not call the storage again, it already has the data from the storage count calls